libr/io/p/io_ar.c

```diff
@@ -0,0 +1,102 @@
+/* radare - LGPLv3 - Copyright 2017 - xarkes */
+#include <r_io.h>
+#include <r_lib.h>
+#include <r_util.h>
+#include <r_cons.h>
+#include "ar.h"
+
+static bool r_io_ar_plugin_open(RIO *io, const char *file, bool many) {
+	if (!strncmp("ar://", file, 5) || !strncmp("lib://", file, 6)) {
+		return true;
+	}
+	return false;
+}
+
+static RIODesc *r_io_ar_open(RIO *io, const char *file, int rw, int mode) {
+	RIODesc *res = NULL;
+	char *arname = strdup (file);
+	arname = strstr (arname, "://");
+	char *filename = strstr(arname + 3, "//");
```

Comment:
Arname heap ptr is lost here. Use a different pointer and free the arname at the end